hazelcast/src/main/java/com/hazelcast/config/TopicConfig.java

```diff
@@ -114,10 +115,44 @@ public boolean isGlobalOrderingEnabled() {
      */
     public TopicConfig setGlobalOrderingEnabled(boolean globalOrderingEnabled) {
         this.globalOrderingEnabled = globalOrderingEnabled;
+        if (this.globalOrderingEnabled && this.multiThreadingEnabled) {
+        	this.multiThreadingEnabled = false;
+        }
         return this;
     }
+    
+    /**
+     * Checks if multi-threaded processing of incoming messages is enabled or not. 
+     * When disabled only one dedicated thread will handle all topic messages. Otherwise
+     * any thread from events thread pool can be used for message handling.
+     *
+     * @return true if multi-threading is enabled, false if disabled
+     */
+    public boolean isMultiThreadingEnabled() {
+        return multiThreadingEnabled;
+    }
+    
 
     /**
+     * Enable multi-threaded message handling. When enabled any thread from events 
+     * thread pool can be used for incoming message processing. Otherwise only one
+     * dedicated thread will be used to handle topic messages.    
+     * Note: it can be enabled only in case when global ordering is disabled.
+     *
+     * @param multiThreadingEnabled set to true to enable multi-threaded message processing, false to disable
+     * @return The updated TopicConfig
+     */
+    public TopicConfig setMultiThreadingEnabled(boolean multiThreadingEnabled) {
+    	if (this.globalOrderingEnabled) {
+    		// log warn/error ?
```

Comment:
you can throw `ConfigurationException`. 
prefer to fail-fast rather than silently "fix" conflicting configuration options.